### PR TITLE
changelog generator tool add `PageableFilter` and `type to any changelog print`

### PR DESCRIPTION
**Pull request by Alancere**
https://github.com/Azure/azure-sdk-for-go/issues/20017

### CODE REVIEW

File: eng/tools/generator/autorest/model/changelog.go [hunk 1]
```diff
@@ -265,8 +269,10 @@ func getSignatureChangeItems(b *report.BreakingChanges) []string {
 		for _, k := range sortChangeItem(b.Structs) {
 			v := b.Structs[k]
 			for f, d := range v.Fields {
-				line := fmt.Sprintf("Type of `%s.%s` has been changed from `%s` to `%s`", k, f, d.From, d.To)
-				items = append(items, line)
+				if !strings.Contains(d.To, "any") {
```
Comment (tadelesh) [inline eng/tools/generator/autorest/model/changelog.go:272]: Could you please remove duplication of these code?

File: eng/tools/generator/autorest/model/changelog.go [hunk 2]
```diff
@@ -419,3 +440,24 @@ func removePattern(funcName string, returnValue string) string {
...
+	if b.IsEmpty() {
+		return nil
+	}
+
+	var items []string
+	if len(b.Structs) > 0 {
+		for _, k := range sortChangeItem(b.Structs) {
+			v := b.Structs[k]
+			for f, d := range v.Fields {
+				if strings.Contains(d.To, "any") {
```
Comment (tadelesh) [inline eng/tools/generator/autorest/model/changelog.go:454]: I'm a little bit struggling about this condition. For most situation, is could be right, but I wonder if there is any corner case that will break it, e.g., `map[A]string` -> `map[B]any`. To be safe, perhaps we only support exact equal at first.

File: eng/tools/generator/cmd/v2/common/changelogProcessor_test.go [hunk 3]
```diff
@@ -79,6 +79,28 @@ func TestLROFilter(t *testing.T) {
 	assert.Equal(t, excepted, changelog.ToCompactMarkdown())
 }
 
+func TestPageableFilter(t *testing.T) {
```
Comment (tadelesh) [inline eng/tools/generator/cmd/v2/common/changelogProcessor_test.go:82]: You may add some tests for "to any" changelog.

File: eng/tools/generator/autorest/model/changelog.go (hunk 1 above)
Comment (Alancere) [reply eng/tools/generator/autorest/model/changelog.go:272]: Duplicate code means duplication with function typeToAny?

Comment (tadelesh) [reply eng/tools/generator/autorest/model/changelog.go:272]: > Duplicate code means duplication with function typeToAny?

Yes.

**Review by tadelesh — APPROVED** [review]

**Pull request merged**